src/api/content.ts

```diff
@@ -87,19 +88,89 @@ async function loadDocument(filePath: string) {
   }
 }
 
-export async function syncContent() {
-  const filePaths = await glob(`${CONTENT_DIR}/*.md`);
+export async function syncContent(contentDir = CONTENT_DIR) {
+  const filePaths = await glob(`${contentDir}/*.md`);
   filePaths.forEach(filePath => {
     taskQueue.pushTask(() => loadDocument(filePath));
   });
-  // TODO: handle deleted files
 }
 
-export function watchContent() {
-  const refreshContentDebounced = debounce(syncContent, 5000, {
-    trailing: true,
-  });
-  fs.watch(CONTENT_DIR, () => {
-    refreshContentDebounced();
-  });
+export function validateDocumentPayload(data: string) {
+  parseBody(data);
+  const fmResult = fm<object>(data);
+
+  const hasAttributes = Object.keys(fmResult.attributes).length;
+  if (!hasAttributes) {
+    throw new Error('invalid empty markdown head file.');
+  }
+
+  const hasBody = fmResult.body.replace(/\r?\n/ig, '').trim();
+  if (!hasBody) {
+    throw new Error('invalid empty markdown body file.');
+  }
+
+  return fmResult;
+}
+
+export function validateDocumentName(file: { originalname: string }) {
+  const hasMDExtension = path.extname(file.originalname) === '.md';
+
+  if (!hasMDExtension) {
+    throw new Error(`can't upload a none markdown file.`);
+  }
+
+  const [, article] = file.originalname.split('.');
+
+  if (!article || article === 'md') {
+    throw new Error(`${file.originalname}: expected filename format <publication>.<article>.md.`);
+  }
+}
+
+export async function removeContentAndTopic(filename: string) {
+  let filePath = path.join(CONTENT_DIR, filename) + '.md';
+
+  let isFileExists = fs.existsSync(filePath);
```

Comment:
The variable name `isFileExists` looks a bit awkward. Perhaps `fileExist` without the `is` prefix reads better here?